scripts/west_commands/sign.py

```diff
@@ -191,88 +193,161 @@ def sign(self, command, build_dir, bcfg, formats):
-            log.inf('Generating:', out_bin)
-            sign_bin = (sign_base + args.tool_args +
-                        [runner_config.bin_file, out_bin])
+            sign_bin = sign_base + [in_bin, out_bin]
+            log.inf('bin: {}'.format(out_bin))
             log.dbg(quote_sh_list(sign_bin))
             subprocess.check_call(sign_bin)
-
-        # Build a signed .hex
-        if 'hex' in formats and runner_config.hex_file:
+        if in_hex:
             out_hex = args.shex or str(b / 'zephyr' / 'zephyr.signed.hex')
-            log.inf('Generating:', out_hex)
-            sign_hex = (sign_base + args.tool_args +
-                        [runner_config.hex_file, out_hex])
+            sign_hex = sign_base + [in_hex, out_hex]
+            log.inf('hex: {}'.format(out_hex))
             log.dbg(quote_sh_list(sign_hex))
             subprocess.check_call(sign_hex)
 
+    @staticmethod
+    def find_imgtool(command, args):
+        if args.tool_path:
+            command.check_force(shutil.which(args.tool_path),
+                                '--tool-path {}: not an executable'.
+                                format(args.tool_path))
+            tool_path = args.tool_path
+        else:
+            tool_path = shutil.which('imgtool')
+            if not tool_path:
+                log.die('imgtool not found; either install it',
+                        '(e.g. "pip3 install imgtool") or provide --tool-path')
+        return tool_path
+
     @staticmethod
     def get_cfg(command, bcfg, item):
         try:
             return bcfg[item]
         except KeyError:
             command.check_force(
-                False,
-                "imgtool parameter unknown, build directory has no {} {}".
-                format('device tree define' if item.startswith('DT_') else
-                       'Kconfig option',
-                       item))
+                False, "build .config is missing a {} value".format(item))
             return None
+
+    @staticmethod
+    def edt_flash_node(b, cache):
+        # Get the EDT Node corresponding to the zephyr,flash chosen DT
+        # node.
+
+        # Retrieve the list of devicetree bindings from cache.
+        try:
+            bindings = cache.get_list('CACHED_DTS_ROOT_BINDINGS')
+            log.dbg('DTS bindings:', bindings, level=log.VERBOSE_VERY)
+        except KeyError:
+            log.die('CMake cache has no CACHED_DTS_ROOT_BINDINGS.'
+                    '\n  Try again after re-building your application.')
+
+        # Ensure the build directory has a compiled DTS file
+        # where we expect it to be.
+        dts = b / 'zephyr' / (cache['CACHED_BOARD'] + '.dts.pre.tmp')
+        if not dts.is_file():
+            log.die("can't find DTS; expected:", dts)
+        log.dbg('DTS file:', dts, level=log.VERBOSE_VERY)
+
+        # Parse the devicetree using bindings from cache.
+        try:
+            edt = edtlib.EDT(dts, bindings)
+        except edtlib.EDTError as e:
```

Comment:
Could also catch `OSError` here and skip the `dts.is_file()` check above.

Reply:
that would result in:

```
FATAL ERROR: can't parse devicetree: [Errno 2] No such file or directory: '/home/mbolivar/zp/zephyr/build/zephyr/nrf52_pca10040.dts.pre.tmp'
```

instead of:

```
FATAL ERROR: can't find DTS; expected: /home/mbolivar/zp/zephyr/build/zephyr/nrf52_pca10040.dts.pre.tmp
```

I think the current error is a bit more user friendly, and I actually want other OSErrors from edtlib to bubble up, since they are unexpected.